go/enclave/components/block_processor.go

```diff
@@ -100,29 +100,27 @@ func (bp *l1BlockProcessor) HealthCheck() (bool, error) {
 func (bp *l1BlockProcessor) tryAndInsertBlock(ctx context.Context, br *common.BlockAndReceipts) (*BlockIngestionType, error) {
 	block := br.Block
 
-	_, err := bp.storage.FetchBlock(ctx, block.Hash())
```

Comment:
I'm not sure this is legitimate? Which block is canonical is determined more than anything by the blocks getting built on top of it, I think we could safely ignore any 'flickering' of head L1 blocks (and not sure if it happens in ethereum anyway).

To me it feels safe for the enclave to only update what is canonical L1 when it receives a **new** L1 block. WDYT?

Reply:
The host submits only canonical blocks into the enclave. That's the contract, afaik.
So this is basically the first block it has to send when it goes back to an old branch

Reply:
Oh ok, so the guardian deals with that currently, so there wasn't a bug. Host only sends canonical blocks, if enclave has already seen it then it moves onto the next one.

```
		if strings.Contains(err.Error(), errutil.ErrBlockAlreadyProcessed.Error()) {
			// we have already processed this block, let's try the next canonical block
			// this is most common when we are returning to a previous fork and the enclave has already seen some of the blocks on it
			// note: logging this because we don't expect it to happen often and would like visibility on that.
			g.logger.Info("L1 block already processed by enclave, trying the next block", "block", block.Hash())
			nextHeight := big.NewInt(0).Add(block.Number(), big.NewInt(1))
			nextCanonicalBlock, err := g.sl.L1Repo().FetchBlockByHeight(nextHeight)
			...
```

My worry is:

1. we introduce some confusion into the contract, because now we don't reject it but we also don't require it.

2. doing the 'ingest' twice on the same block might be a route for bugs in the future, it was quite easy to reason about when we could say the enclave would **never** process a block with the same hash twice.

Reply:
What happened before was that the next block built on top of the "previous head/previous noncanonica/current head" block would be regarded as a "fork" with a "canonical" back-chain. 
That would update the batches accordingly.
It was just doing that with a delay.

Reply:
Yeah fair enough. I think we should get rid of that block of logic from the guardian then and agree from now on that the enclave will expect blocks to be resubmitted if they become canonical again.

Reply:
I think that's the behaviour of the etheruem rpc. 
But ,the good thing is that the enclave does not depend on that. It can handle either case, as long as it has all the blocks